### ISSUE TITLE
In browser Pendulum account

### ISSUE THREAD

**Issue by prayagd** [pendulum-chain/pendulum-webapp#65]
Problem statement
Currently the webApp user cant setup his Pendulum account

User Story
1. As a pendulum webApp user, I should be see all my pendulum assets on the dashboard. 
2. Currently the user is able to setup a stellar wallet using the “one click setup”. From now on the user should be able to setup a Pendulum account in the browser itself or he should be able to connect a external browser wallet like polkdot.js or talismann. 

Acceptance Criteria
The webApp user should be able to create a pendulum account and see the balances on the dashboard.

**Comment by ashneverdawn** [comment]
Hmm, Seems like using ed25519 as native addresses onPendulum is going to be a common issue for us when integrating with the rest of the Dot/Sama ecosystem. 

Perhaps it would be easier to go with sr25519 on Pendulum? This would mean we'd need to implement a mapping on Pendulum between the sd25519 and ed25519 adresses. Probably an extrinsic to 'register' the mapping. If they are both generated from the same seed phrase, it could still be seamless for the user.

Seems to me it will be a lot simpler to solve it this way - we only have to solve the conversion for Stellar instead of for every other wallet/parachain out there. Not to mention potentially xcm.

wdyt? @gonzamontiel  @TorstenStueber

**Comment by gonzamontiel** [comment]
@TorstenStueber this is something we already discussed at the very beginning of Pendulum, and it's becoming more relevant now. If the reasoning above is correct, we should proceed and create a ticket for it.

**Comment by TorstenStueber** [comment]
My understanding is that Substrate does neither have "ed25519 addresses" or "sr25519 addresses" but that it is totally agnostic about this. Adresses are just 32 bytes of binary data.

When adding a signature to a Substrate transaction (i.e., extrinsic), it is marked whether the signature is meant to be an ed25519 or sr25519 (or ecdsa signature). Only then is the address interpreted as either an ed25519 public key or as an sr25519 public key.

Of course a wallet will attach more meaning to addresses as they are either created as public ed25519 keys or as public sr25119 keys (each time generated from some secret seed).

Currently we make the wallet ourselves so I don't quite understand this problem. Can you elaborate?

Anyway, if we need to use sr25519 key pairs, then we can either
- always derive both an sr25519 and an ed25519 from the same seed phrase so that every user has two different account addresses in Pendulum
- or change the API of our bridge so that an _issue_ always takes a target address (on Pendulum) as an extra argument, i.e., we never assume that a user wants to move tokens from a Stellar address to the same Pendulum address but to any configurable Pendulum address instead.

What do you think?

**Comment by TorstenStueber** [comment]
Anyway, I think that this ticket here is totally unrelated to es25519 vs sr25519 key pair issues.

**Comment by gonzamontiel** [comment]
Hey. For the sake of the thread.

> If we need to use sr25519 key pairs, then we can either
> 
> always derive both an sr25519 and an ed25519 from the same seed phrase so that every user has two different account ids in Pendulum
> or change the API of our bridge so that an issue always takes a target account id (on Pendulum) as an extra argument, i.e., we never assume that a user wants to move tokens from a Stellar address to the same Pendulum address but to any configurable Pendulum address instead.

We had some discussions regarding this and agreed in your second approach. It is the most intuitive way that the user specifies an address to _bridge to_, in both ways (stellar <-> pendulum). This will unblock this ticket. I created a [preliminary ticket](https://github.com/pendulum-chain/spacewalk/issues/68) on the Spacewalk repo to achieve this, but is low priority by now.

Nevertheless, the ticket needs a more precise specification, @prayagd since we have some time here, can you precisely describe what flow would you expect, provide mockups, etc?

**Comment by prayagd** [comment]
Closing. This is not relevant anymore as we completed SBP-M2.
